### ISSUE TITLE
Syntax error when using this library with PHP 5.6

### ISSUE THREAD

**Issue by sbhalim** [ramsey/uuid#221]
I got syntax error when I was using this library with PHP 5.6.33.

```
PHP Parse error:  syntax error, unexpected '->' (T_OBJECT_OPERATOR) in ..../vendor/ramsey/uuid/src/FeatureSet.php on line 259
```

It seems that PHP parser pre v7 is inconsistent including handling of ```->``` operator.

https://stackoverflow.com/questions/41980469/php5-assignment-operation-result-not-treated-as-object-by-parser

Since the library claims to be compatible with PHP 5.4+, some expressions need to be rewritten to be safe with PHP before v7.

A quick grep (on uuid v3.7.3) give these lines:

```
ramsey/uuid//src/Codec/TimestampFirstCombCodec.php:70:        return $this->getBuilder()->build($this, $this->getFields($fivePieceComponents));
ramsey/uuid//src/Codec/GuidStringCodec.php:70:        return $this->getBuilder()->build($this, $this->getFields($components));
ramsey/uuid//src/Uuid.php:618:        return self::getFactory()->fromBytes($bytes);
ramsey/uuid//src/Uuid.php:629:        return self::getFactory()->fromString($name);
ramsey/uuid//src/Uuid.php:640:        return self::getFactory()->fromInteger($integer);
ramsey/uuid//src/Uuid.php:676:        return self::getFactory()->uuid1($node, $clockSeq);
ramsey/uuid//src/Uuid.php:689:        return self::getFactory()->uuid3($ns, $name);
ramsey/uuid//src/Uuid.php:699:        return self::getFactory()->uuid4();
ramsey/uuid//src/Uuid.php:712:        return self::getFactory()->uuid5($ns, $name);
ramsey/uuid//src/FeatureSet.php:259:        return (new RandomGeneratorFactory())->getGenerator();
ramsey/uuid//src/FeatureSet.php:279:        ))->getGenerator();
```

**Comment by ramsey** [comment]
Thanks for reporting this.

From your grep output, it looks like the only places affected by the issue described on Stack Overflow are lines 259 and 279 in the FeatureSet class. I’ll take a look and make sure this is backwards-compatible syntax.

It’s odd that the [tests are passing on Travis CI](https://travis-ci.org/ramsey/uuid/builds/357447937) (without syntax errors) for all versions of PHP tested.

**Comment by sbhalim** [comment]
@ramsey Sure I just cloned the project and the tests are passed even on my computer. I'll try to come up with reproducible script.

**Comment by ramsey** [comment]
Thanks. I would be surprised if this were the case, but maybe none of the tests cover the FeatureSet class?

**Comment by sbhalim** [comment]
Well after some digging up, it appears that the problem was actually my test framework.

I used [Kahlan](https://github.com/kahlan/kahlan) to test my project. I just realised that it does some just-in-time transformations on the file during autoload that change the line to be something like ```return ((new $__KMONKEY__46()))->getGenerator();```. It's still a bug, but it's PHP's, not this project's. 

Excluding Uuid namespace from Kahlan JIT solves it. I will close this.
